app/controllers/wpcc/engine_controller.rb

```diff
@@ -1,7 +1,22 @@
 module Wpcc
   class EngineController < Wpcc.parent_controller.constantize
     protect_from_forgery
+    before_action :check_session
 
     layout 'wpcc/engine'
+
+    def check_session
+      session_expired? ? reset_session : update_session_expiry
+    end
+
+    private
+
+    def session_expired?
+      session[:expires_at] && session[:expires_at] <= Time.current
+    end
+
+    def update_session_expiry
+      session[:expires_at] = Time.current + 30.minutes
```

Comment:
@norm has made a good point about the brevity of the session expiry config. Shall we have a quick catch up to vote on a sensible figure, please?